lib/build.ts

```diff
@@ -90,20 +90,36 @@ const generateLabels = (task: BuildTask): { labels?: Dictionary<string> } => {
  * @param docker The handle to the docker daemon
  * @return a promise which resolves to a LocalImage which points to the produced image
  */
-export function runBuildTask(
+export async function runBuildTask(
 	task: BuildTask,
 	docker: Dockerode,
 	registrySecrets: RegistrySecrets,
 	secrets?: SecretsPopulationMap,
 	buildArgs?: Dictionary<string>,
 ): Promise<LocalImage> {
-	// First merge in the registry secrets (optionally being
-	// overridden by user input) so that they're available for
-	// both pull and build
-	task.dockerOpts = _.merge(
-		{ registryconfig: registrySecrets },
-		task.dockerOpts,
-	);
+	// check if docker supports propagating which platform to build for
+	// NOTE: docker api version 1.34 actually introduced platform to the
+	// api but it was broken until fixed in
+	// https://github.com/moby/moby/commit/7f334d3acfd7bfde900e16e393662587b9ff74a1
+	// which is why we check for 1.38 here
+	const usePlatformOption: boolean =
+		task.dockerPlatform &&
+		semver.satisfies(
+			semver.coerce((await docker.version()).ApiVersion),
+			'>=1.38.0',
```

Comment:
Build is failing here:
```
lib/build.ts:105:8 - error TS2322: Type 'boolean | "" | undefined' is not assignable to type 'boolean'.
  Type 'undefined' is not assignable to type 'boolean'.

105  const usePlatformOption: boolean =
           ~~~~~~~~~~~~~~~~~

lib/build.ts:108:4 - error TS2345: Argument of type 'SemVer | null' is not assignable to parameter of type 'string | SemVer'.
  Type 'null' is not assignable to type 'string | SemVer'.

108  		semver.coerce((await docker.version()).ApiVersion),
     		~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I think this solves it:
```
	const usePlatformOption: boolean =
		!!task.dockerPlatform &&
		semver.satisfies(
			semver.coerce((await docker.version()).ApiVersion) || '0.0.0',
			'>=1.38.0',
		);
```
(Note I also had to add `!!` in front of `task.dockerPlatform`.)

Reply:
thanks!